### PR TITLE
Pushing iso3dfd_omp_offload code sample

### PR DESCRIPTION
**Pull request by slgogar**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Pushing iso3dfd_omp_offload code sample and minor changes to iso3dfd_dpcpp code sample.

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [X] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [X] Implement coding guidelines and ensure code quality.
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [X] Adhere to readme template 
- [X] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [X] Enforce format via clang-format config file
- [X] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [x] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [x] Tested using Dev Cloud when applicable
- [x] Implement fixes for ONSAM Jiras
- [X] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: DirectProgramming/DPC++/StructuredGrids/iso3dfd_omp_offload/include/iso3dfd.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
+//==============================================================
+// Copyright © 2020 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <omp.h>
+#include <chrono>
```
[the remark this answers is not in the document]
Comment (pmpeter1) [reply DirectProgramming/DPC++/StructuredGrids/iso3dfd_omp_offload/include/iso3dfd.h:8]: Agreed as written dpc_commom.hpp is only usabl;e by DPC++ samples.  We would need to create an omp_common.hpp for OpenMP programs that does not pull in extra DPC++ dependences.

[the remark this answers is not in the document]
Comment (JoeOster) [reply DirectProgramming/DPC++/StructuredGrids/iso3dfd_omp_offload/include/iso3dfd.h:8]: Iis that a suggestion or something that needs to be done? Perhaps for Beta10?

[the remark this answers is not in the document]
Comment (pmpeter1) [reply DirectProgramming/DPC++/StructuredGrids/iso3dfd_omp_offload/include/iso3dfd.h:8]: We need to plan on how to support OpenMP shared headers for Beta10.

[the remark this answers is not in the document]
Comment (JoeOster) [reply DirectProgramming/DPC++/StructuredGrids/iso3dfd_omp_offload/include/iso3dfd.h:8]: Approving this for now, will re-address the shared headers for openMP in Beta 10

**Comment by slgogar** [comment]
@pmpeter1 @JoeOster I updated some variables as per your recommendation. Also changed use of #define MIN to omp declare inline function. However i have still kept the #define for STENCIL_LOOKUP and STENCIL_LOOKUP_Z to brevity.. Most of the RTM codes use a similar approach for stencil computation.

**Comment by JoeOster** [comment]
@slgogar - please take a look at the error thrown by ci

**Comment by srdontha** [comment]
@slgogar @JoeOster The OMP sample should go to the corresponding C++ folder per this URL. Can you please ensure this gets fixed ASAP?  https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C++/StructuredGrids/iso3dfd_omp_offload

**Comment by JoeOster** [comment]
@anjgola - As this is just a json error, I recommend approving and merging. When the .json is updated with "icpx" or a different path is chosen we can fix at that time

**Comment by anjgola** [comment]
Hi @JoeOster , I just saw this message. Unfortunately the system cannot bypass json changes so even though this was merged it is causing sample aggregator to fail and not run. As I had said in the emails the schema should have been fixed at the same time if this was going to get merged. I will send an email to address this.

**Pull request merged**